src/rt/access.rs

```diff
@@ -14,6 +14,11 @@ impl Access {
         }
     }
 
+    pub(crate) fn set(&mut self, path_id: usize, version: &VersionVec) {
```

Comment:
I think it is not possible to do directly without sacrificing much of the benefit - to use `clone_from` we must have an existing instance of `Access` and creating it at the call site of `set_last_access` incurs an additional unnecessary allocation.

One possible way to implement it is to store and update `Access` directly in the `Thread` struct but it feels unnatural (and storing `path_id` can be error-prone as it can get out of sync with the actual path position).

Reply:
Ah, yeah, you're right; the `clone_from` behavior is not identical. Nevermind!